packages/core/src/hooks/useLocalStorage.ts

```diff
@@ -29,17 +30,20 @@ function setItem(key: string, value: any) {
  * @internal Intended for internal use - use it on your own risk
  */
 export function useLocalStorage(key: string) {
-  const [value, setValue] = useState(() => getItem(key))
+  const {
+    localStorageOverride = typeof window !== 'undefined' ? window.localStorage : global.localStorage,
+  } = useConfig()
+  const [value, setValue] = useState(getItem(key, localStorageOverride))
```

Comment:
```suggestion
  const [value, setValue] = useState(() => getItem(key, localStorageOverride))
```

Preserve previous behaviour.